### ISSUE TITLE
using empty GSUFFIX, MSUFFIX, USUFFIX

### ISSUE THREAD

**Issue by sgparry** [martymac/ldapscripts#14]
Empty suffixes in the /etc/ldapscripts/ldapscripts.conf lead to generation of illegal dns in the LDIF.

```
Dec 13 01:12:47 host2020 ldapscripts: ldapaddgroup(xxxxxxx): /usr/local/sbin/ldapaddgroup xxxxxx 1004
SASL/EXTERNAL authentication started
SASL username: gidNumber=0+uidNumber=0,cn=peercred,cn=external,cn=auth
SASL SSF: 0
ldap_add: Invalid DN syntax (34)
        additional info: invalid DN
  -> Error adding group xxxxxx to LDAP
```

**Comment by martymac** [comment]
Hello @sgparry,

Thanks for your report.

The scripts have not been designed to work without dedicated suffixes for each type of entry (user, group, machine). Adding that possibility would required a major rewrite which is not planned at the moment.

Best regards,

Ganael.

**Comment by sgparry** [comment]
I was able to hack a workaround by reordering the substitutions and adding one to replace double dot with single dot. It's not pretty but it works - I will post a diff later if that helps... 

On 16 December 2021 16:46:04 GMT, Ganael Laplanche ***@***.***> wrote:
>Closed #14.
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/martymac/ldapscripts/issues/14#event-5779688974
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>
-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.
